### PR TITLE
Replaced inline JavaScript in wrapper

### PR DESCRIPTION
**Pull request by J0WI**
#3849 rewritten

---
### How to test
- add a new wrapper
- see that the iframe looks good in frontend
- appy this patch
- make sure it works still expected :smile:

### CODE REVIEW

File: media/com_wrapper/index.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<!DOCTYPE html><title></title>
```
Comment (brianteeman) [inline media/com_wrapper/index.html]: We no longer require an empty index.html file in every folder

File: media/com_wrapper/js/index.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<!DOCTYPE html><title></title>
```
Comment (brianteeman) [inline media/com_wrapper/js/index.html]: We no longer require an empty index.html file in every folder

**Comment by brianteeman** [comment]
It would really help if you cold provide some explanation what this does and how to test it. The easier it is to test the quicker it will get merged<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5716">issues.joomla.org/joomla-cms/5716</a>.</sub>

**Comment by J0WI** [comment]
As described in #3838, inline JS prevents admins to use a restrictive CSP on their site. So I moved this JS snippet, that controls the height of an iframe, to a file.

**Comment by dgt41** [comment]
@J0WI Thanks for coding this
Can I ask you why you decided not to use this:
```php
JFactory::getApplication()->getDocument()->addScriptDeclaration("
CODE GOES HERE
");
```
This way there is no need for the 2 extra javascript files, and also no extra request on the rendered file.
The weight is small only 12 lines of code… 
You can also see that in my attempt to do the same thing here: #5117

**Comment by J0WI** [comment]
This function would create a ```<script>``` element, isn't it? So it prevents you from using CSP without ```unsafe-inline```. 
https://developer.mozilla.org/en-US/docs/Web/Security/CSP/CSP_policy_directives#Keywords

**Comment by dgt41** [comment]
@J0WI It will add a `<script>***</script>` in the head of the document, so CSP should be fine with it!

**Comment by J0WI** [comment]
Nope, referring to spec all ```<script>``` elements are blocked, even if they are created in head. Just verified using nginx and Firefox 37.

**Comment by dgt41** [comment]
@J0WI WOW really? Joomla then needs more tweaks to comply with this 😃

**Comment by J0WI** [comment]
That would be great! The backed is actually unusable with CSP. D:
JFYI: eval is evil #3837

**Comment by dgt41** [comment]
@J0WI And I proposed few days ago something that uses evil() #5652 😕

**Comment by dgt41** [comment]
@J0WI Just out of curiosity the 'self' will reject the inline scripts from the same source? Documentation is not very clear...

**Comment by J0WI** [comment]
self => allowed only from exact same url/protocol (e.g. http**s**://github.com/* but nothing else, relative paths are allowed)
unsafe-inline => allow creepy inline stuff (e.g. <script> element or javascript: values in href)
unsafe-eval => allow evil eval() stuff

The same restrictions are available for nearly every resource, so inline CSS is also a problem. But you can't do that much damage as with JS.

You may be interested in https://developer.mozilla.org/en-US/docs/Web/Security/CSP/Using_Content_Security_Policy#Examples.3A.C2.A0Common_use_cases

**Comment by phproberto** [comment]
This is the right way :+1:

**Comment by J0WI** [comment]
@brianteeman Do you need any additional information from me to merge this? Is it already to late to push this to 3.4?

**Comment by dgt41** [comment]
@J0WI There where some proposal for wrapper by Hannes #5117 (see the last comments). Do you think you can implement these in this PR?

**Comment by J0WI** [comment]
Done.

**Comment by J0WI** [comment]
This still waits for a review/merge

**Comment by zero-24** [comment]
Adding test Instructions.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5716">issues.joomla.org/joomla-cms/5716</a>.</sub>

**Comment by dgt41** [comment]
@zero-24 This needs to be tested also for the module
#test ok here!

**Comment by zero-24** [comment]
Thanks @dgt41 just tested with the component and module works good here. Thanks @J0WI moving to RTC :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/5716">issues.joomla.org/joomla-cms/5716</a>.</sub>

**Comment by Kubik-Rubik** [comment]
Thank you @J0WI! Merged with https://github.com/joomla/joomla-cms/commit/5e44e0a80df88001c26168155258cf3317bb7d6f

**Pull request closed without merging**